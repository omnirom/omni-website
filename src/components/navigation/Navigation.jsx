import './_nav.scss';
import './_social-media-button.scss';
import _ from 'lodash';
import React, {Component} from 'react';
import PropTypes from 'prop-types';

export class Navigation extends Component {
  handleClick(routeType, event) {
    const {
      handleNavRoute,
      navItemClick
    } = this.props;

    if (routeType === 'internal') {
      handleNavRoute(routeType, event.target.getAttribute('href').slice(1), event);
    }

    navItemClick();
  }

  render() {
    const {links} = this.props;

    return (
      <ul className={`${this.props.hasOwnProperty('listClassname') ? this.props.listClassname : ''} nav`}>
        {
          links.map((link, index) => (
            <li className="nav__list-item" key={index}>
              <a
                className={`${link.hasOwnProperty('anchorClassname') ? link.anchorClassname : ''} nav__item-link`}
                href={link.url}
                target={link.target || '_self'}
                onClick={this.handleClick.bind(this, link.routeType)}
              >{link.label}</a>
            </li>
          ))
        }
      </ul>
    );
  }
}

Navigation.defaultProps = {
  navItemClick: _.noop
};

Navigation.propTypes = {
  handleNavRoute: PropTypes.func,
  links: PropTypes.array,
  listClassname: PropTypes.string,
  navItemClick: PropTypes.func,
  push: PropTypes.func
};

export default Navigation;
