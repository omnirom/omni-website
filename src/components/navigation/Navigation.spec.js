import React from 'react';
import {Navigation} from './Navigation.jsx';
import renderer from 'react-test-renderer';

describe('Navigation Component', () => {
  const navItems = [
    {url: '#', label: 'Nav One'},
    {url: '#', label: 'Nav Two'},
    {url: '#', label: 'Nav Three'},
    {url: '#', label: 'Nav Four'},
    {url: '#', label: 'Nav Five'}
  ];

  it('renders navigation', () => {
    const component = renderer.create(
      <Navigation
        links={navItems}
        listClassname="test-nav"
      />);
    const tree = component.toJSON();

    expect(tree).toMatchSnapshot();
  });
});
