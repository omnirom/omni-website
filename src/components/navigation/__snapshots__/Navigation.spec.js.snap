// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navigation Component renders navigation 1`] = `
<ul
  className="test-nav nav"
>
  <li
    className="nav__list-item"
  >
    <a
      className=" nav__item-link"
      href="#"
      target="_self"
    >
      Nav One
    </a>
  </li>
  <li
    className="nav__list-item"
  >
    <a
      className=" nav__item-link"
      href="#"
      target="_self"
    >
      Nav Two
    </a>
  </li>
  <li
    className="nav__list-item"
  >
    <a
      className=" nav__item-link"
      href="#"
      target="_self"
    >
      Nav Three
    </a>
  </li>
  <li
    className="nav__list-item"
  >
    <a
      className=" nav__item-link"
      href="#"
      target="_self"
    >
      Nav Four
    </a>
  </li>
  <li
    className="nav__list-item"
  >
    <a
      className=" nav__item-link"
      href="#"
      target="_self"
    >
      Nav Five
    </a>
  </li>
</ul>
`;
